compiler/rustc_session/src/options.rs

```diff
@@ -970,7 +970,7 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
     mir_emit_retag: bool = (false, parse_bool, [TRACKED],
         "emit Retagging MIR statements, interpreted e.g., by miri; implies -Zmir-opt-level=0 \
         (default: no)"),
-    mir_opt_level: usize = (1, parse_uint, [TRACKED],
+    mir_opt_level: Option<usize> = (None, parse_opt_uint, [TRACKED],
```

Comment:
I don't see a reason for making this an Option, it doesn't change its default dependent on coverage like symbol mangling does.

Reply:
I don't know of any other way to know if the option was set on the command line or not.

I need to know that in config.rs when checking for conflicting command line flags.

Reply:
Ah, I understand your point... Let me think about this a minute. I'm inclined to agree.

Reply:
To me the value matters more than if it was overridden (which you know it was today if it's > 1).. if the default _were_ changed to 2, we would still want this to fire to warn people who were using coverage. (Probably we'd either keep the default as 1 if coverage were enabled, or add support for inlining counters. But it's unlikely that we turn on inlining by default anytime soon.)

Reply:
Sorry, yes, I concur. I'll remove the option from mir_opt_level. Thanks!

Reply:
> if the default _were_ changed to 2, we would still want this to fire to warn people who were using coverage.

I agree 100%. Thanks.